cmd/k8sm-executor/main.go

```diff
@@ -54,135 +48,120 @@ var (
-		log.Info("No api servers specified.")
-	} else {
-		if apiClient, err := getApiserverClient(); err != nil {
-			log.Errorf("Unable to make apiserver client: %v", err)
-		} else {
-			// Send events to APIserver if there is a client.
-			log.Infof("Sending events to APIserver.")
-			record.StartRecording(apiClient.Events(""), "kubelet")
-		}
+	if err := util.ApplyOomScoreAdj(*oomScoreAdj); err != nil {
+		log.Info(err)
+	}
+
+	kcfg := standalone.KubeletConfig{
+		Address:                 address,
+		AuthPath:                *authPath,
+		ApiServerList:           apiServerList,
+		AllowPrivileged:         *allowPrivileged,
+		HostnameOverride:        *hostnameOverride,
+		RootDirectory:           *rootDirectory,
+		NetworkContainerImage:   *networkContainerImage,
+		SyncFrequency:           *syncFrequency,
+		RegistryPullQPS:         *registryPullQPS,
+		RegistryBurst:           *registryBurst,
+		MinimumGCAge:            *minimumGCAge,
+		MaxContainerCount:       *maxContainerCount,
+		ClusterDomain:           *clusterDomain,
+		ClusterDNS:              clusterDNS,
+		Port:                    *port, // TODO(nnielsen): Don't hardwire port, but use port from resource offer.
+		CAdvisorPort:            *cAdvisorPort,
+		EnableServer:            true,
+		EnableDebuggingHandlers: *enableDebuggingHandlers,
+		DockerClient:            util.ConnectToDockerOrDie(*dockerEndpoint),
+		EtcdClient:              kubelet.EtcdClientOrDie(etcdServerList, *etcdConfigFile),
 	}
 
-	// Log the events locally too.
-	record.StartLogging(log.Infof)
+	driver := new(mesos.MesosExecutorDriver)
 
-	capabilities.Initialize(capabilities.Capabilities{
-		AllowPrivileged: *allowPrivileged,
-	})
+	//HACK(jdef) largely hacked from the k8s standalone package, createAndInitKubelet
+	initialized := make(chan struct{})
+	builder := standalone.KubeletBuilder(func(kc *standalone.KubeletConfig) (standalone.KubeletBootstrap, *kconfig.PodConfig) {
+		pc := kconfig.NewPodConfig(kconfig.PodConfigNotificationSnapshotAndUpdates)
+		updates := pc.Channel(MESOS_CFG_SOURCE)
+
+		// TODO(k8s): block until all sources have delivered at least one update to the channel, or break the sync loop
+		// up into "per source" synchronizations
+
+		k := &kubeletExecutor{
+			Kubelet: kubelet.NewMainKubelet(
+				kc.Hostname,
+				kc.DockerClient,
+				kc.EtcdClient,
+				kc.RootDirectory,
+				kc.NetworkContainerImage,
+				kc.SyncFrequency,
+				float32(kc.RegistryPullQPS),
+				kc.RegistryBurst,
+				kc.MinimumGCAge,
+				kc.MaxContainerCount,
+				pc.SeenAllSources,
+				kc.ClusterDomain,
+				net.IP(kc.ClusterDNS)),
+			driver:      driver,
+			initialized: initialized,
+		}
+		driver.Executor = executor.New(driver, k.Kubelet, updates, MESOS_CFG_SOURCE)
 
-	dockerClient, err := docker.NewClient(getDockerEndpoint())
-	if err != nil {
-		log.Fatal("Couldn't connnect to docker.")
-	}
+		log.V(2).Infof("Initialize executor driver...")
+		driver.Init()
 
-	hostname := getHostname()
+		k.BirthCry()
```

Comment:
Perhaps it reflects their excitement and optimism in bringing a new kubelet into the world.